test/ParallelIOTest.cpp

```diff
@@ -140,6 +140,60 @@ TEST_CASE( "hdf5_write_test", "[parallel][hdf5]" )
 
     o.flush();
 }
+
+TEST_CASE( "hdf5_write_test_zero_extent", "[parallel][hdf5]" )
+{
+    int mpi_s{-1};
+    int mpi_r{-1};
+    MPI_Comm_size(MPI_COMM_WORLD, &mpi_s);
+    MPI_Comm_rank(MPI_COMM_WORLD, &mpi_r);
+    uint64_t size = static_cast<uint64_t>(mpi_s);
+    uint64_t rank = static_cast<uint64_t>(mpi_r);
+    Series o = Series("../samples/parallel_write_zero_extent.h5", AccessType::CREATE, MPI_COMM_WORLD);
+
+    ParticleSpecies& e = o.iterations[1].particles["e"];
+
+    /* every rank n writes n consecutive cells, increasing values
+     * rank 0 does a zero-extent write
+     * two ranks will result in {1}
+     * three ranks will result in {1, 2, 3}
+     * four ranks will result in {1, 2, 3, 4, 5, 6} */
+    uint64_t num_cells = ((size-1)*(size-1) + (size-1))/2; /* (n² + n) / 2 */
```

Comment:
hm, maybe use `n^2` here to stay in full-ascii source? had troubles with some GCC versions in the past with non-ascii chars in source files.

Reply:
CLion was *smart* enough to make it superscript.